packages/client/src/sync/fetcher/accountfetcher.ts

```diff
@@ -218,6 +154,129 @@ export class AccountFetcher extends Fetcher<JobTask, AccountData[], AccountData>
     )
   }
 
+  async fetch(): Promise<boolean> {
+    this.fetcherDoneFlags.syncing = true
+
+    try {
+      this.fetcherDoneFlags.accountFetcher.started = true
+      const accountFetch = !this.fetcherDoneFlags.accountFetcher.done ? super.fetch() : null
+      // wait for all accounts to fetch else storage and code fetcher's doesn't get us full data
+      this.config.superMsg(`Snapsync: running accountFetch=${accountFetch !== null}`)
+      await accountFetch
+      this.fetcherDoneFlags.accountFetcher.started = false
+      if (this.fetcherDoneFlags.accountFetcher.done !== true) {
+        // @scorbajio need to see the reason for this here
```

Comment:
I collected logs for several test runs but I have not yet been able to reproduce this error case. If there is a better way to reproduce it, let me know. Or if you can reproduce it, it would help if you could share logs for a full run.